### PR TITLE
Update UI (login menu and button) based on login state / enable logout

### PR DESCRIPTION
**Pull request by chanseokoh**
- Updates the UI message of the login menu and the tooltip of the login toolbar button based on the current login state.
- Now users can log out, with a logout confirmation dialog.

One gripe is that the same messages are duplicated both in `plugin.properties` and `messages.properties` for some localized messages (for the initial menu name and tooltip). It would be nice if `plugin.xml` could pull values directly from `messages.properties`.

### CODE REVIEW

File: plugins/com.google.cloud.tools.eclipse.appengine.login/src/com/google/cloud/tools/eclipse/appengine/login/GoogleLoginCommandHandler.java [hunk 1]
```diff
@@ -1,21 +1,63 @@
...
+        "com.google.cloud.tools.eclipse.appengine.login.commands.loginCommand", null); //$NON-NLS-1$
 
-    MessageDialog.openInformation(shell,
-        "TESTING AUTH", success ? "SUCCESS" : "FAILURE (to be implemented)");
     return null;
   }
+
+  @Override
+  public void updateElement(UIElement element, @SuppressWarnings("rawtypes") Map parameters) {
+    boolean loggedIn = new GoogleLoginService().getCachedActiveCredential() != null;
```
Comment (briandealwis) [inline plugins/com.google.cloud.tools.eclipse.appengine.login/src/com/google/cloud/tools/eclipse/appengine/login/GoogleLoginCommandHandler.java]: When you get to the point of implementing an OSGi service, you can use `UIElement#getServiceLocator()`

File: plugins/com.google.cloud.tools.eclipse.appengine.login/src/com/google/cloud/tools/eclipse/appengine/login/GoogleLoginService.java [hunk 2]
```diff
@@ -127,6 +127,14 @@ Credential createCredentialHelper(String accessToken, String refreshToken) {
     return credential;
   }
 
+  public void clearCredential() {
+    MApplication application = PlatformUI.getWorkbench().getService(MApplication.class);
+
+    Map<String, Object> synchronizedMap =
```
Comment (briandealwis) [inline plugins/com.google.cloud.tools.eclipse.appengine.login/src/com/google/cloud/tools/eclipse/appengine/login/GoogleLoginService.java]: Hmm, using a synchronizeMap doesn't do anything here — other accessors won't be going through your map, and other callers to this method will get a new synchronizedMap.

If you're really concerned, put an `AtomicReference`

Comment (chanseokoh) [inline plugins/com.google.cloud.tools.eclipse.appengine.login/src/com/google/cloud/tools/eclipse/appengine/login/GoogleLoginService.java]: > other callers to this method will get a new synchronizedMap.

Ah, I see. I may have misunderstood how sychronizedMap can work. Then I should re-work this thing in the other methods too. (Not in this PR.) I think for now I'll remove the synchronizedMap here, and do something about the re-work in another PR.

File: plugins/com.google.cloud.tools.eclipse.appengine.login/src/com/google/cloud/tools/eclipse/appengine/login/GoogleLoginCommandHandler.java (hunk 1 above)
Comment (chanseokoh) [inline plugins/com.google.cloud.tools.eclipse.appengine.login/src/com/google/cloud/tools/eclipse/appengine/login/GoogleLoginCommandHandler.java]: Note taken. Thanks.

**Comment by briandealwis** [comment]
LGTM.  We should have some tests for the login state, but that can wait for the integration tests.

**Pull request merged**